content/en/docs/concepts/storage/volumes.md

```diff
@@ -942,358 +808,66 @@ Simultaneous writers are not allowed.
-A free developer license is available.
+## Volume Plugin Interfaces
 
-{{< caution >}}
-You must run the StorageOS container on each node that wants to
-access StorageOS volumes or that will contribute storage capacity to the pool.
-For installation instructions, consult the
-[StorageOS documentation](https://docs.storageos.com).
-{{< /caution >}}
+The Out-of-tree volume plugins include the Container Storage Interface (CSI)
+and FlexVolume. They enable storage vendors to create custom storage plugins. The main Kubernetes binary implements the CSI, and communicates with any external plugins. This keeps the main binary efficent, and allows users to choose their own storage plugins. 
 
-The following example is a Pod configuration with StorageOS:
+Before the introduction of CSI and FlexVolume, all volume plugins (like
+volume types listed above) were "in-tree" meaning they were built, linked,
+compiled, and shipped with the core Kubernetes binaries and extend the core
+Kubernetes API. This meant that adding a new storage system to Kubernetes (a
+volume plugin) required checking code into the core Kubernetes code repository.
 
-```yaml
-apiVersion: v1
-kind: Pod
-metadata:
-  labels:
-    name: redis
-    role: master
-  name: test-storageos-redis
-spec:
-  containers:
-    - name: master
-      image: kubernetes/redis:v1
-      env:
-        - name: MASTER
-          value: "true"
-      ports:
-        - containerPort: 6379
-      volumeMounts:
-        - mountPath: /redis-master-data
-          name: redis-data
-  volumes:
-    - name: redis-data
-      storageos:
-        # The `redis-vol01` volume must already exist within StorageOS in the `default` namespace.
-        volumeName: redis-vol01
-        fsType: ext4
-```
+Both CSI and FlexVolume allow volume plugins to be developed independent of
+the Kubernetes code base, and deployed (installed) on Kubernetes clusters as
+extensions.
 
-For more information about StorageOS, dynamic provisioning, and PersistentVolumeClaims, see the
-[StorageOS examples](https://github.com/kubernetes/examples/blob/master/volumes/storageos).
+For storage vendors looking to create an out-of-tree volume plugin, please refer
+to the [volume plugin FAQ](https://github.com/kubernetes/community/blob/master/sig-storage/volume-plugin-faq.md).
 
-### vsphereVolume {#vspherevolume}
+### csi
 
 {{< note >}}
-You must configure the Kubernetes vSphere Cloud Provider. For cloudprovider
-configuration, refer to the [vSphere Getting Started guide](https://vmware.github.io/vsphere-storage-for-kubernetes/documentation/).
-{{< /note >}}
-
-A `vsphereVolume` is used to mount a vSphere VMDK volume into your Pod.  The contents
-of a volume are preserved when it is unmounted. It supports both VMFS and VSAN datastore.
+CSI drivers may not be compatible across all Kubernetes releases.
+Please check the specific CSI driver's documentation for supported
+deployments steps for each Kubernetes release and a compatibility matrix.
```

Comment:
This looks importamt; I would wrap this in a ` {{< note >}}` &hellip;  `{{< note >}}`

Reply:
It looks like it is to me - unless you are referring to something besides the CSI drivers.